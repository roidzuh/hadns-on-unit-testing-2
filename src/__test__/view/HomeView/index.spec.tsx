import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import HomeView from '@/view/Home';

describe('home view', () => {
  it('should handle render homepage', () => {
    const { container } = render(<HomeView />);
    expect(container).toMatchSnapshot();
  });

  it('should handle button click', () => {
    const { getByText } = render(<HomeView />);
    const button = getByText('submit');
    fireEvent.click(button);
    expect(button).toBeInTheDocument();
  });
});
