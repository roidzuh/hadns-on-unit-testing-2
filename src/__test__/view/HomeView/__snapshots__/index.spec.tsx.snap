// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home view should handle render homepage 1`] = `
<div>
  <div>
    <h1>
      add a user
    </h1>
    <div>
      <label
        for="Name"
      >
        Name
      </label>
      <input
        id="Name"
        name="Name"
        type="text"
      />
    </div>
    <div>
      <label
        for="Email"
      >
        Email
      </label>
      <input
        id="Email"
        name="Email"
        type="text"
      />
    </div>
    <button>
      submit
    </button>
    <div>
      <label
        for="Description"
      >
        Description
      </label>
      <textarea
        id="Description"
        name="Description"
      />
    </div>
    <hr />
    <h1>
      list of user
    </h1>
    <table>
      <thead>
        <tr>
          <th>
            name
          </th>
          <th>
            email
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            jane
          </td>
          <td>
            jane@email.com
          </td>
        </tr>
        <tr>
          <td>
            alex
          </td>
          <td>
            alex@email.com
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
